plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -403,21 +392,15 @@ private Optional<SystemTable> getRawSystemTable(ConnectorSession session, Schema
                 // Handled above.
                 break;
             case HISTORY:
-                if (name.getSnapshotId().isPresent()) {
-                    throw new TrinoException(NOT_SUPPORTED, "Snapshot ID not supported for history table: " + systemTableName);
-                }
                 return Optional.of(new HistoryTable(systemTableName, table));
             case SNAPSHOTS:
-                if (name.getSnapshotId().isPresent()) {
-                    throw new TrinoException(NOT_SUPPORTED, "Snapshot ID not supported for snapshots table: " + systemTableName);
-                }
                 return Optional.of(new SnapshotsTable(systemTableName, typeManager, table));
             case PARTITIONS:
-                return Optional.of(new PartitionTable(systemTableName, typeManager, table, getSnapshotId(table, name.getSnapshotId(), isAllowLegacySnapshotSyntax(session))));
+                return Optional.of(new PartitionTable(systemTableName, typeManager, table, Optional.empty()));
```

Comment:
This looks like a regression to me, does the `AS OF` syntax work as a replacement for `select * from abc$manifests@456`?

Reply:
I have a PR for supporting time travel queries on system tables but it is unfortunately not ready yet.

Reply:
Probably it is best to introduce the ability to perform time travel queries on system tables before continuing with this PR.

Reply:
As shown in the PRs: 

- https://github.com/trinodb/trino/pull/13497
- https://github.com/trinodb/trino/pull/13854

time travel on system tables incurs added complexity to the Iceberg connector code.

Without concrete demand from the community for time travel functionality on system tables, we leave the above mentioned PRs on hold and go further with disabling support  for snapshot id within the Iceberg table name.

Reply:
> This looks like a regression to me

correct

> does the `AS OF` syntax work as a replacement for `select * from abc$manifests@456`?

no

The reason we're _not_ pursing the above linked PRs is because we believe there is little community interest in these.
If there isn't general demand on time travel on system tables, we should be OK with carrying out removal of this (already deprecated) functionality.